### PR TITLE
Wenyi5608 chatglm

### PR DESCRIPTION
**Pull request by wenyi5608**
Add instructions for running on windows

### CODE REVIEW

**Comment by wenyi5608** [comment]
It is duplicated, I closed #42 . Thank you!

File: llm/chatglm_cpp/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+$copy <OpenVINO dir>\runtime\bin\intel64\Release\* build\llm\chatglm_cpp\Release\
+$copy <OpenVINO dir>\runtime\3rdparty\tbb\bin\tbb12.dll build\llm\chatglm_cpp\Release\
+```
+
+copy user_ov_extensions.dll and its dependent dll to chatglm.exe directory
+```sh
+$copy build\thirdparty\openvino_contrib\modules\custom_operations\user_ie_extensions\Release\user_ov_extensions.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\third_party\lib\icudt70.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\third_party\lib\icuuc70.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\lib\core_tokenizers.dll build\llm\chatglm_cpp\Release\
```
Comment (ilya-lavrenov) [inline llm/chatglm_cpp/README.md:47]: currently, during the build stage all dependencies are also copied to folder with `user_ov_extensions.dll`. So, we can copy from a single place.

Comment (wenyi5608) [reply llm/chatglm_cpp/README.md:47]: I compiled with the latest code but the fast_tokenizer-src dll(icudt70.dll, icuuc70.dll and core_tokenizers.dll) are not found in the user_ov_extensions.dll directory.

File: llm/chatglm_cpp/chatglm.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	    BATCH_SIZE, -1
+        }}
+    };
+
+    
+    std::vector<ov::Output<ov::Node>> inputs = model->inputs();
+    
+    for (size_t idx = 3; idx < inputs.size(); ++idx) {
+        ov::PartialShape shape = inputs.at(idx).get_partial_shape();
+        shape[1] = BATCH_SIZE;
```
[the remark this answers is not in the document]
Comment (wenyi5608) [reply llm/chatglm_cpp/chatglm.cpp:63]: If this approach is adopted, running different LLM models requires recompiling the code. Is it better to do LLM model selection through parameter configuration?

[the remark this answers is not in the document]
Comment (FionaZZ92) [reply llm/chatglm_cpp/chatglm.cpp:63]: From user's expectation, the llm sample should align with llama.cpp, chatglm.cpp, qwen.cpp... Cause users normally pick one model/sample to refer. No one will adopt a single large main.cpp to adopt all LLM models. So, I suggest if we want to combine. Can we just refer to cpp samples just put each xxx.cpp as sub-project into one solution, just built them together, but run exe separately?

[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply llm/chatglm_cpp/chatglm.cpp:63]: agree @FionaZZ92 approach. We can compile the same code twice to have `llm` and `llm-chatglm` apps.
Second app will additionally define a variable to override BATCH_IDX to 1 (default is 0).

File: README.md [hunk 3]
```diff
@@ -7,11 +7,46 @@ Every pipeline requires https://github.com/openvinotoolkit/openvino for C++ to b
...
+
+```sh
+$copy build\thirdparty\openvino_contrib\modules\custom_operations\user_ie_extensions\Release\user_ov_extensions.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\third_party\lib\icudt70.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\third_party\lib\icuuc70.dll build\llm\chatglm_cpp\Release\
+$copy build\_deps\fast_tokenizer-src\lib\core_tokenizers.dll build\llm\chatglm_cpp\Release\
+```
+Usage: `chatglm.exe <openvino_model.xml> <tokenizer.xml> <detokenizer.xml> "device"`
+
+Example: `build\llm\chatglm_cpp\Release\chatglm.exe .\chatglm3-6b\openvino_model.xml .\tokenizer.xml .\detokenizer.xml "CPU"`
```
Comment (ilya-lavrenov) [inline README.md:50]: I suppose chatglm specific things must be in llm/chatglm_cpp/README.md only

File: README.md [hunk 4]
```diff
@@ -7,11 +7,46 @@ Every pipeline requires https://github.com/openvinotoolkit/openvino for C++ to b
 
 Build the pipelines and `user_ov_extensions`
 
+1.Clone submodules:
 ```sh
+git clone https://github.com/wenyi5608/openvino.genai.git -b wenyi5608-chatglm
```
[the remark this answers is not in the document]
Comment (wenyi5608) [reply README.md]: I update the README.md, and delete custom repo.

File: README.md (hunk 3 above)
Comment (wenyi5608) [reply README.md]: I update the README.md, and put the chatglm instruction guide in the chatglm directory.

**Comment by ilya-lavrenov** [comment]
let's resolve merge conflicts and move sample to https://github.com/openvinotoolkit/openvino.genai/tree/master/text_generation/

Please, also add GHA workflow here https://github.com/openvinotoolkit/openvino.genai/tree/master/.github/workflows to ensure the sample complies and runs

**Comment by Wovchena** [comment]
chatglm was added by https://github.com/openvinotoolkit/openvino.genai/pull/97

**Pull request closed without merging**